server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStats.java

```diff
@@ -0,0 +1,108 @@
+        builder.startObject()
+            .field(Fields.SHARD_ID, remoteSegmentUploadShardStats.shardId)
+
+            .field(Fields.LOCAL_REFRESH_TIMESTAMP, remoteSegmentUploadShardStats.localRefreshTimeMs)
+            .field(Fields.LOCAL_REFRESH_CUMULATIVE_COUNT, remoteSegmentUploadShardStats.localRefreshCount)
+            .field(Fields.REMOTE_REFRESH_TIMESTAMP, remoteSegmentUploadShardStats.remoteRefreshTimeMs)
+            .field(Fields.REMOTE_REFRESH_CUMULATIVE_COUNT, remoteSegmentUploadShardStats.remoteRefreshCount)
+            .field(Fields.BYTES_LAG, remoteSegmentUploadShardStats.bytesLag)
+
+            .field(Fields.REJECTION_COUNT, remoteSegmentUploadShardStats.rejectionCount)
+            .field(Fields.CONSECUTIVE_FAILURE_COUNT, remoteSegmentUploadShardStats.consecutiveFailuresCount);
+
+        builder.startObject(Fields.TOTAL_REMOTE_REFRESH);
+        builder.field(Fields.STARTED, remoteSegmentUploadShardStats.totalUploadsStarted)
+            .field(Fields.SUCCEEDED, remoteSegmentUploadShardStats.totalUploadsSucceeded)
+            .field(Fields.FAILED, remoteSegmentUploadShardStats.totalUploadsFailed);
+        builder.endObject();
+
+        builder.startObject(Fields.TOTAL_UPLOADS_IN_BYTES);
+        builder.field(Fields.STARTED, remoteSegmentUploadShardStats.uploadBytesStarted)
+            .field(Fields.SUCCEEDED, remoteSegmentUploadShardStats.uploadBytesSucceeded)
+            .field(Fields.FAILED, remoteSegmentUploadShardStats.uploadBytesFailed);
+        builder.endObject();
+        builder.startObject(Fields.REMOTE_REFRESH_SIZE_IN_BYTES);
+        builder.field(Fields.LAST_SUCCESSFUL, remoteSegmentUploadShardStats.lastSuccessfulRemoteRefreshBytes);
+        builder.field(Fields.MOVING_AVG, remoteSegmentUploadShardStats.uploadBytesMovingAverage);
+        builder.endObject();
+
+        builder.startObject(Fields.UPLOAD_LATENCY_IN_BYTES_PER_SEC);
+        builder.field(Fields.MOVING_AVG, remoteSegmentUploadShardStats.uploadBytesPerSecMovingAverage);
+        builder.endObject();
+        builder.startObject(Fields.REMOTE_REFRESH_LATENCY_IN_NANOS);
+        builder.field(Fields.MOVING_AVG, remoteSegmentUploadShardStats.uploadTimeMovingAverage);
+        builder.endObject();
+        builder.endObject();
+
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalWriteable(remoteSegmentUploadShardStats);
+    }
+
+    static final class Fields {
+        static final String SHARD_ID = "shard_id";
+        static final String LOCAL_REFRESH_TIMESTAMP = "local_refresh_timestamp_in_millis";
+        static final String LOCAL_REFRESH_CUMULATIVE_COUNT = "local_refresh_cumulative_count";
+        static final String REMOTE_REFRESH_TIMESTAMP = "remote_refresh_timestamp_in_millis";
+        static final String REMOTE_REFRESH_CUMULATIVE_COUNT = "remote_refresh_cumulative_count";
```

Comment:
this is not really count. It can happen that the remote can jump from 1 to 10 on account of continuous upload failures followed by a success. This might seem confusing to users in such case where the jump is sudden in a very short span of time.

Reply:
Why do we need to increment on retries as well?

Reply:
it does not get incremented on retry. it is related to the number of refreshes behind (considering every refresh is assigned a unique number - not the retries)

Reply:
does it make sense to just send out the diff/lag here ? i think sending two fields might be confusing